Storage/container/one2one.h

```diff
@@ -0,0 +1,253 @@
+/*******************************************************************************
+The MIT License (MIT)
+
+Copyright (c) 2016 Grigory Nikolaenko <nikolaenko.grigory@gmail.com>
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*******************************************************************************/
+
+#ifndef CURRENT_STORAGE_CONTAINER_ONE2ONE_H
+#define CURRENT_STORAGE_CONTAINER_ONE2ONE_H
+
+#include "common.h"
+#include "sfinae.h"
+
+#include "../base.h"
+
+#include "../../TypeSystem/optional.h"
+#include "../../Bricks/util/comparators.h"  // For `CurrentHashFunction`.
+
+namespace current {
+namespace storage {
+namespace container {
+
+template <typename T,
+          typename T_UPDATE_EVENT,
+          typename T_DELETE_EVENT,
+          template <typename...> class ROW_MAP,
+          template <typename...> class COL_MAP>
+class GenericOne2One {
+ public:
+  using T_ROW = sfinae::ENTRY_ROW_TYPE<T>;
+  using T_COL = sfinae::ENTRY_COL_TYPE<T>;
+  using T_RCPAIR = std::pair<T_ROW, T_COL>;
```

Comment:
+1 for `T_KEY` (or `key_t`, as we're about to slowly transition to this syntax). I never really liked `T_RCPAIR` =)